### ISSUE TITLE
`DirectInvocationOnMock` doesn't understand `MockedStatic` with `@Mock`

### ISSUE THREAD

**Issue by DenWav** [google/error-prone#4039]
The `DirectInvocationOnMock` inspection warns when calling methods on [`MockedStatic`](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/MockedStatic.html) objects such as `when` or `verify`, if the `MockedStatic` is declared using [`@Mock`](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mock.html).

Example:

```java
import org.junit.jupiter.api.BeforeEach;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
class ExampleTest {
    @Mock
    HttpClient mockedClient;

    @Mock
    MockedStatic<HttpClient> staticHttpClient;

    @BeforeEach
    void setup() {
        this.staticHttpClient.when(HttpClient::newHttpClient).thenReturn(this.mockedClient);
    }
}
```

This code snippet will generate a `DirectInvocationOnMock` warning for the `staticHttpClient.when()` call, but that is correct usage of the API.

**Comment by gregallen** [comment]
Similar issue for `MockedConstruction`
